### PR TITLE
Added documentation into NovelRT::Ecs.

### PR DESCRIPTION
**Pull request by RubyNova**

### CODE REVIEW

File: include/NovelRT/Ecs/ComponentBuffer.h [hunk 1]
```diff
@@ -17,6 +26,13 @@ namespace NovelRT::Ecs
         T _deleteInstructionState;
 
         public:
+        /**
+         * @brief Constructs a new ComponentBuffer for type T.
+         * 
+         * 
+         * @param poolSize The amount of worker threads being utilisied in this instance of the ECS.
```
Comment (capnkenny) [inline include/NovelRT/Ecs/ComponentBuffer.h:33]: Spelling mistake: utilising

File: include/NovelRT/Ecs/EcsUtils.h [hunk 2]
```diff
@@ -10,7 +10,15 @@ namespace NovelRT::Ecs
...
     using EntityId = Atom;
     using ComponentTypeId = Atom;
-   
+    
+    /**
+     * @brief Retrieves the ComponentTypeId for a given component type.
+     * 
+     * This free function will create a new ID for the given type if one does not exist on the first call.
+     * 
+     * @tparam TComponent The component type to get he ID for.
```
Comment (capnkenny) [inline include/NovelRT/Ecs/EcsUtils.h:19]: Spelling mistake: the

File: include/NovelRT/Ecs/EntityCache.h [hunk 3]
```diff
@@ -6,25 +6,51 @@
...
+        /**
+         * @brief Queues an entity for removal in the next frame.
+         * 
+         * @param poolId The current worker thread's pool ID the removal instruction is coming from.
+         * @param entityToRemove The EntityId to delete.
+         */
         void RemoveEntity(size_t poolId, EntityId entityToRemove) noexcept;
 
+        /**
+         * @brief Propagates deletion requests from worker threads to the main thrad. Once this is called, the data can be accessed from EntityCache::GetEntitiesToRemoveThisFrame.
```
Comment (capnkenny) [inline include/NovelRT/Ecs/EntityCache.h:50]: Spelling mistake: thread

**Review by capnkenny — CHANGES_REQUESTED** [review]
Looks good! A couple spelling mistakes need fixing before this can be approved.

File: include/NovelRT/Ecs/ComponentBuffer.h (hunk 1 above)
Comment (capnkenny) [reply include/NovelRT/Ecs/ComponentBuffer.h:33]: Sorry - `utilised`

Comment (RubyNova) [reply include/NovelRT/Ecs/ComponentBuffer.h:33]: What's the typo here?

Comment (RubyNova) [reply include/NovelRT/Ecs/ComponentBuffer.h:33]: Oh nevermind

**Review by capnkenny — APPROVED** [review]

**Pull request merged**